src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -1137,6 +1138,30 @@ R"(#include <stdio.h>
         last_expr_precedence = 16;
     }
 
+    void visit_StringConcat(const ASR::StringConcat_t &x) {
+    if (is_strcat_util_func_added){
+        additional_functions =
+            "char *concat (const char *str1, const char *str2) {\n"
+            "\n"
+            "size_t len1 = strlen (str1);         /* get lenghts */\n"
+            "size_t len2 = strlen (str2);\n"
+            "\n"
+            "char * s = malloc (len1 + len2 + 1); /* allocate s and concat with ' ' */\n"
+            "memcpy (s, str1, len1);\n"
```

Comment:
I was thinking if we can add all the helper functions like this or like in #1109 in some other file and include that header file.  Something similar to `lfortran_intrinsics.h` or do we want some other new file for this? @certik @czgdp1807

Reply:
I agree all these functions fit well in `lfortran_intrinsics.h`. That way no need to define dynamically in C backend.